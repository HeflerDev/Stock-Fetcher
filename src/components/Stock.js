import React from 'react';
import PropTypes from 'prop-types';

const Stock = ({ stock }) => {
  const {
    symbol,
    name,
    currency,
    stockExchange,
    exchangeShortName,
  } = stock;

  return (
    <div className="stack">
      <ul className="board">
        <li className="col-12 big">
          <div>{ name }</div>
        </li>
        <li className="col-3">
          <div>{ symbol }</div>
        </li>
        <li className="col-3">
          <div>{ currency }</div>
        </li>
        <li className="col-3">
          { stockExchange }
        </li>
        <li className="col-3">
          { exchangeShortName }
        </li>
      </ul>
    </div>
  );
};

Stock.propTypes = {
  stock: PropTypes.shape({
    symbol: PropTypes.string.isRequired,
    name: PropTypes.string,
    currency: PropTypes.string.isRequired,
    stockExchange: PropTypes.string,
    exchangeShortName: PropTypes.string,
  }).isRequired,
};

Stock.defaultProps = {
  stock: {
    name: 'None',
  },
}

export default Stock;
