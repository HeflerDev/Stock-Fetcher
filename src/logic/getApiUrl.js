const apiKey = 'af6d5d2d48987d8ac328c48e5bf40e8e';

const getApiUrl = {
  profile(value) {
    return `https://financialmodelingprep.com/api/v3/profile/${value}?apikey=${apiKey}`;
  },
  search(value) {
    return `https://financialmodelingprep.com/api/v3/search?query=${value}&limit=10&apikey=${apiKey}`;
  },
  actives() {
    return `https://financialmodelingprep.com/api/v3/actives?apikey=${apiKey}`;
  },
  gainers() {
    return `https://financialmodelingprep.com/api/v3/gainers?apikey=${apiKey}`;
  },
  losers() {
    return `https://financialmodelingprep.com/api/v3/losers?apikey=${apiKey}`;
  },
};

export default getApiUrl;
