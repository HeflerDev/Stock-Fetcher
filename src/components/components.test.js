import React from 'react';
import 'react-router-dom';
import renderer from 'react-test-renderer';
import Stock from './Stock';
import getApiUrl from '../logic/getApiUrl';

it('renders Stock Correctly', () => {
  const tree = renderer
    .create(<Stock stock={
      {
        symbol: 'AAPL',
        name: 'Apple',
        currency: 'USD',
        exchangeShortName: 'NASDAQ',
      }
    }
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('when rendering the page', () => {
  const url = getApiUrl.profile('AAPL');
  async function getProfileObj() {
    return fetch(url)
      .then((res) => {
        if (res.ok) { return res.json(); }
        throw new Error();
      })
      .then((data) => data);
  }
  it('somting', async () => {
    expect.assertions(1);
    const [data] = await getProfileObj();
    expect(typeof data).toBe('object');
  });
});
