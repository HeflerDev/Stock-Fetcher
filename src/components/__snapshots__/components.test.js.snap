// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Stock Correctly 1`] = `
<div
  className="stack"
>
  <ul
    className="board"
  >
    <li
      className="col-12 big"
    >
      <div>
        Apple
      </div>
    </li>
    <li
      className="col-4"
    >
      <div>
        AAPL
      </div>
    </li>
    <li
      className="col-4"
    >
      <div>
        USD
      </div>
    </li>
    <li
      className="col-4"
    >
      NASDAQ
    </li>
  </ul>
</div>
`;
